### PR TITLE
Add libsodium

### PR DESCRIPTION
**Pull request by yamashi**

### CODE REVIEW

File: packages/l/libsodium/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        add_deps("autoconf", "automake", "libtool", "pkg-config")
+        set_urls("https://download.libsodium.org/libsodium/releases/libsodium-$(version)-stable.tar.gz")
+        add_versions("1.0.18", "91441b13c965e241cd64bb42823f1f2d882f08f8f40f1716cda17df2b3450af2")
+    elseif is_plat("mingw") then
+        set_urls("https://download.libsodium.org/libsodium/releases/libsodium-$(version)-stable-mingw.tar.gz")
+        add_versions("1.0.18", "d2f1918e198cd86b9e6ba05b2f5c2dc86753875ea3ee887892767231c6b7e121")
+    end
+
+
+
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua:19]: Please remove the extra blank lines and keep only one line

File: packages/l/libsodium/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    on_load(function (package)
+        if not package:config("shared") then
+            package:add("defines", "SODIUM_STATIC")
+        end
+    end)
+    
+    on_install("windows", function (package)
+        os.cp("include", package:installdir())
+        os.cp(path.join((is_arch("x86") and "Win32" or "x64"), (package:debug() and "Debug" or "Release"), "v142", (package:config("shared") and "dynamic" or "static"), "*"), package:installdir("lib"))
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua:28]: package:is_plat

File: packages/l/libsodium/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        package:add("links", "libsodium.lib")
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua:29]: xmake will find all libraries in lib directory as links.  Unless necessary, such as adjusting the link order, there is usually no need to manually configure them

File: packages/l/libsodium/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        
+    end)
+
+    on_install("mingw", function (package)
+
+        local root_dir = (is_arch("x86") and "libsodium-win32" or "libsodium-win64")
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua:35]: package:is_plat

File: packages/l/libsodium/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+        os.cp(path.join(root_dir, "include"), package:installdir())
+        if package:config("shared") then
+            os.cp(path.join(root_dir, "lib", "libsodium.dll.a"), package:installdir("lib"))
+            os.cp(path.join(root_dir, "bin", "*"), package:installdir("lib"))
+            package:add("links", "libsodium.dll.a")
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua:41]: we need not add links.

File: packages/l/libsodium/xmake.lua [hunk 6]
```diff
@@ -23,12 +23,12 @@ package("libsodium")
     
     on_install("windows", function (package)
         os.cp("include", package:installdir())
-        os.cp(path.join((package:is_plat("x86") and "Win32" or "x64"), (package:debug() and "Debug" or "Release"), "v142", (package:config("shared") and "dynamic" or "static"), "*"), package:installdir("lib"))
+        os.cp(path.join((package:is_plat("x64") and "x64" or "Win32"), (package:debug() and "Debug" or "Release"), "v142", (package:config("shared") and "dynamic" or "static"), "*"), package:installdir("lib"))
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua:26]: package:is_arch("x64")     not is_plat

File: packages/l/libsodium/xmake.lua [hunk 7]
```diff
@@ -23,12 +23,12 @@ package("libsodium")
...
     end)
 
     on_install("mingw", function (package)
 
-        local root_dir = (package:is_plat("x86") and "libsodium-win32" or "libsodium-win64")
+        local root_dir = (package:is_plat("x64") and "libsodium-win64" or "libsodium-win32")
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua:31]: package:is_arch("x86_64")

File: packages/l/libsodium/xmake.lua [hunk 8]
```diff
@@ -23,12 +23,12 @@ package("libsodium")
     
     on_install("windows", function (package)
         os.cp("include", package:installdir())
-        os.cp(path.join((package:is_plat("x64") and "x64" or "Win32"), (package:debug() and "Debug" or "Release"), "v142", (package:config("shared") and "dynamic" or "static"), "*"), package:installdir("lib"))
+        os.cp(path.join((package:is_arch("x86_64") and "x64" or "Win32"), (package:debug() and "Debug" or "Release"), "v142", (package:config("shared") and "dynamic" or "static"), "*"), package:installdir("lib"))
```
Comment (waruqi) [inline packages/l/libsodium/xmake.lua]: x64 not x86_64

windows/msvc: x86, x64
mingw: i386, x86_64

**Comment by waruqi** [comment]
Maybe it's because I still don't know how to use squash merge. I think it might be better for you to merge and submit directly.

```console
git reset --soft xxx
git commit -a -m "add libsodium"
git push origin SteamSockets --force
```

**Comment by yamashi** [comment]
image

**Comment by waruqi** [comment]
Oh, this is a great feature, thank you for telling me.

**Pull request merged**